components/service/glean/docs/pings/custom.md

```diff
@@ -32,9 +53,12 @@ type, you can add the special value `default` to `send_in_pings`:
 
 ## Sending a custom ping
 
-To send a custom ping, use `Glean.sendPings`. It is up to the application to
-call this method at the schedule and cadence that is appropriate for the ping.
+To send a custom ping, call the `send` method on the `PingType` object that
+glean generated for your ping.
+
+For example, to send the custom ping defined above:
 
 ```kotlin
-Glean.sendPings(listOf("search"))
+import org.mozilla.yourApplication.GleanMetrics.Pings
+Pings.search.send()
```

Comment:
Does this buy us much more over the `Glean.sendPing` API? I'd avoid going through the hassle of generating new files (potential binary overhead?) to only wrap `Glean.sendPings(listOf("search"))`. Moreover, I think this is more of a product direction question that should involve @georgf (and also FYI @badboy )

Reply:
We need metadata about whether a particular ping should contain `client_id` stored somewhere.  There are two cases for this:
- When sending events that have been queued from a previous run at app startup.  If we don't have the ping metadata, we don't know whether any events in a custom ping should include a client_id in that ping.
- When using the debug activity, so we don't require the tester to get that right.
The first pass at implementing this included `includeClientId` only as a parameter to `sendPings`, which wasn't good enough because it could support these use cases.

I think this will buy us a lot in the future as well as pings grow more options and functionality.

I wouldn't worry about binary overhead.  This should always be a very small number of small data classes.

Reply:
Good points. I think I'm hooked into this now :)

Reply:
I like how this exposes the `.send` function through the API of the ping instead of the Glean singleton and doesn't rely on the user to know the String name of the ping.